simulator/mips/t/unit_test.cpp

```diff
@@ -2006,6 +2235,55 @@ TEST_CASE( "MIPS64_instr: srlv 0x11 by 0x00000a00 (shift-variable overflow)")
     CHECK( instr.get_v_dst() == 0x11);
 }
 
+TEST_CASE( "MIPS32_instr: sub")
+{
+    CHECK(MIPS32Instr(0x01398822).get_disasm() == "sub $s1, $t1, $t9");
+    MIPS32Instr instr( "sub");
+    instr.set_v_src( 1, 0);
+    instr.set_v_src( 1, 1);
+    instr.execute();
+    CHECK(instr.get_v_dst() == 0);
+    
+    instr.set_v_src( 10, 0);
+    instr.set_v_src( 1, 1);
+    instr.execute();
+    CHECK(instr.get_v_dst() == 9);
+    
+    instr.set_v_src( 1, 0);
+    instr.set_v_src( 0, 1);
+    instr.execute();
+    CHECK(instr.get_v_dst() == 1);
+    
+    /*  Overflow exception is not implemented (#130)
+    instr.get_v_dst( 0xfee1dead);
+    instr.set_v_src( 0x80000000, 0);
+    instr.set_v_src( 0xffffffff, 1);
+    instr.execute();
+    CHECK(instr.get_v_dst() == 0xfee1dead);
+    CHECK(instr.trap_type() != Trap::NO_TRAP);
+    */
+}
+
+TEST_CASE( "MIPS32_instr: subu")
+{
+    CHECK(MIPS32Instr(0x01398823).get_disasm() == "subu $s1, $t1, $t9");
+    MIPS32Instr instr( "subu");
+    instr.set_v_src( 1, 0);
+    instr.set_v_src( 1, 1);
+    instr.execute();
+    CHECK(instr.get_v_dst() == 0);
+    
```

Comment:
Please separate the different test cases to different TEST_CASE macro.
We should not re-use entities between different unit tests, e.g. call `execute` twice on the same instruction.

Reply:
Ok, I hope I understood it correctly.